### PR TITLE
The final Milestone for the Capstone Project

### PR DESCRIPTION
**Pull request by fitz95**
## Group Capstone Project
- Use JavaScript to make websites dynamic and build basic single-page apps.
- Use ES6 syntax.
- Use ES6 modules.
- Use callbacks and promises.
- Use webpack.
- Apply JavaScript best practices and language style guides in code.
- Use AAA pattern for unit tests.
- Write units tests for a JavaScript app.
- Follow Gitflow.
- Solve simple git conflicts.
- Send and receive data from an API.
- Use API documentation.
- Understand and use JSON.
- Make JavaScript code asynchronous.

### CODE REVIEW

File: src/modules/api.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      },
+      body: JSON.stringify({
+        item_id: `${ides[index]}`,
+      }),
+    });
+    return userPost;
+  } catch (error) {
+    return error;
+  }
+};
```
Comment (rloterh) [inline src/modules/api.js:94]: - [ ] OPTIONAL
Great job, _like_ feature properly implemented! However, I believe we can improve the UX of our app by adding an _unlike_ feature also, to enable a user easily revert a _like_ action that is unintended.

File: src/modules/commentpopup.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  popupContent.appendChild(heading);
+  const detailDiv = document.createElement('div');
+  detailDiv.className = 'detail-flex';
+  popupContent.appendChild(detailDiv);
+  const areaP = document.createElement('p');
+  areaP.className = 'detailstyle';
+  areaP.innerHTML = `Area:    ${meal.meals[0].strArea}`;
+  const paragraph = document.createElement('p');
+  // if (meal.meals[0].strInstructions.length > 500) {
+  //   meal.meals[0].strInstructions = meal.meals[0].strInstructions.substring(0,500) + '...';
```
Comment (rloterh) [inline src/modules/commentpopup.js]: - [ ] OPTIONAL
There are commented code blocks found in your code on the repository. Kindly endeavor to remove **all** commented code from your code before pushing it to the repository so as to keep your code well organized.

File: src/modules/api.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import ui from './ui.js';
+import resBtn from './reservation.js';
+
+const id = 'tKVlvnEbmf4TMWB77SE7';
+const urlMealrecipe = 'https://www.themealdb.com/api/json/v1/1/lookup.php?i=';
+const urlAllMeals = 'https://www.themealdb.com/api/json/v1/1/filter.php?c=Seafood';
+const invApiUrl = `https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/${id}/likes`;
```
Comment (rloterh) [inline src/modules/api.js:7]: - Persisting and retrieving record(s) with the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) well implemented! 👌

File: src/modules/com.test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/**
+ * @jest-environment jsdom
```
Comment (rloterh) [inline src/modules/com.test.js:2]: - Nice job on the unit test implementation! 👌

File: README.md [hunk 5]
```diff
@@ -19,37 +10,39 @@
...
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-none
+You would be able see later a Live Version of this App Below
+ https://fitz95.github.io/API-BASED-APP-/
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-
 ## 💻 Getting Started <a name="getting-started"></a>
```
Comment (rloterh) [inline README.md:46]: - [ ] OPTIONAL
Kindly include appropriate instructions under the _Getting Started_ section, especially on how to _run_ and _test_ your app.

**Review by rloterh — APPROVED** [review]
Hi Team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: src/modules/api.js (hunk 1 above)
Comment (mohisa302) [reply src/modules/api.js:94]: Dear @rloterh , actually we wanted to do this also in the first, but the API that [Microverse template](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) gave us for this project doesn't have a POST request to decrease or delete likes values, we only can send the item id in request and it increases it by one unit, we appreciate any suggestion about how we can do this 🙏

File: src/modules/commentpopup.js (hunk 2 above)
Comment (mohisa302) [reply src/modules/commentpopup.js]: done

File: README.md (hunk 5 above)
Comment (mohisa302) [reply README.md:46]: done

File: src/modules/api.js (hunk 1 above)
Comment (rloterh) [reply src/modules/api.js:94]: Hi @mohisa302 

I recall couple of other students implement the _unlike_ feature, This project is fairly new and I haven't worked on it myself yet. Meanwhile let look into it and I'll get back to you. 

Cheers! 😃

**Pull request merged**